specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json

```diff
@@ -623,7 +623,7 @@
           }
         }
       },
-      "put": {
+      "patch": {
```

Comment:
What is your goal here? Are you just trying to mitigate creation? You could return bad request from put if the resource wasn't pre-existing.

If you want to stick with patch however you have to follow:
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#patch-resource
It will require you to allow partial resource updates as all properties in a patch are optional and should be joined with what's in storage.

Reply:
The goal is to allow specific InvoiceSection or BillingProfile property. PUT to PATCH is made to fix the documentation error since these APIs are only supporting patch and put will throw an error. Yes we support partial update through patch in InvoiceSection and BillingProfile update. User is allowed to updated only desired InvoiceSection or BillingProfile property.

Reply:
If this is the case, then you need to change the body parameter to not include readonly or required properties. (Id, Name, Type... cannot be updated in patch, so the model shouldn't include them)

Here a more descriptive comment from a Microsoft.Billing PR last week: (https://github.com/Azure/azure-rest-api-specs/pull/6041)

Patch typically doesn't require the same input as Put. You should create a new type definition which contains only the 'patchable' properties of a BillingAccount, the convention we typically use would name this new definition 'BillingAccountUpdateParameters'.

Take a look at the follow swagger for an example, search for StorageAccountUpdateParameters.
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storage/resource-manager/Microsoft.Storage/stable/2018-11-01/storage.json

For more info on how Patch should work:
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#patch-resource

Reply:
Do you plan to support Put/re-put in the future? You're prohibiting people from using templates to manipulate invoices and profiles by only supporting patch.

Reply:
You are right, we need to support Put in future. I am reverting this change for now since this requires few more actions from our end. Will make those code changes and then send a new swagger PR.